pkg/apps/utils.tsx

```diff
@@ -27,14 +27,21 @@ import { show_modal_dialog } from "cockpit-components-dialog.jsx";
 
 const _ = cockpit.gettext;
 
-export function icon_url(path_or_url) {
+export function icon_url(path_or_url: string): string {
     if (!path_or_url)
         return "default.png";
 
     if (path_or_url[0] != '/')
         return path_or_url;
 
-    const queryobj = {
+    type QueryObj = {
```

Comment:
I think `interface` is preferred over `type`. Can't say why, however.

Reply:
Oh, ok -- in my mental model `type` is  similar to `struct` (i.e. data only), and `interface` similar to `class` (i.e. behaviour/methods). We use `type` a lot for shaping data, e.g. in pkg/lib/cockpit/channel.ts, but that also has e.g. `interface BaseChannelOptions` which is also data only The `interface`s in pkg/lib/cockpit.d.ts are all class-y (with methods).

https://blog.logrocket.com/types-vs-interfaces-typescript/ shows some difference, like unions (type only) or merging and "subclassing" with `extends` (interface only).

So, I don't have a strong opinion on that, and right now we don't have much consistency here. I'm fine with moving to `interface`.

Reply:
Yeah, thanks for investigating. I had this paragraph from the TS Handbook in mind:

For the most part, you can choose based on personal preference, and TypeScript will tell you if it needs something to be the other kind of declaration. If you would like a heuristic, use interface until you need to use features from type.

https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#interfaces

I guess we can always change a type to an interface without affecting the users, so... *shrug* :-)

Reply:
> If you would like a heuristic, use interface until you need to use features from type.

That's a good piece of advice, thanks!